src/main/java/com/hashedin/bitly/ShortCodeMappingService.java

```diff
@@ -0,0 +1,63 @@
+package com.hashedin.bitly;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ShortCodeMappingService {
+
+	private final Map<String, URI> mapping;
+	
```

Comment:
This maps short codes to a fully qualified URI. We could have stored the destination as a String, but a URI makes it easier to validate and to be sure we don't have bad data.

Also notice final keyword. The final keyword, along with the unmodifiableMap below, makes our class immutable. Once constructed, there is no way to change the data. 

Immutable classes have many advantages - the most important being they are thread safe.